src/System.Device.Gpio/Interop/Unix/libgpiod/V1/SafeLineHandle.cs

```diff
@@ -7,31 +7,53 @@
 namespace System.Device.Gpio.Libgpiod.V1;
 
 /// <summary>
-/// Pointer to a pin.
+/// Pointer to a pin (Not a real SafeLineHandle, because we need to align its finalization with the owning object)
 /// </summary>
-internal class SafeLineHandle : SafeHandle
+internal sealed class SafeLineHandle : IDisposable
 {
-    public PinMode PinMode { get; set; }
-
-    public SafeLineHandle()
-        : base(IntPtr.Zero, true)
+    private IntPtr _handle;
+    public SafeLineHandle(IntPtr handle)
     {
+        _handle = handle;
+        PinMode = PinMode.Input;
     }
 
-    protected override bool ReleaseHandle()
+    public PinMode PinMode { get; set; }
+
+    public IntPtr Handle
     {
-        // Contrary to intuition, this does not invalidate the handle (see comment on declaration)
-        LibgpiodV1.gpiod_line_release(handle);
-        return true;
+        get
+        {
+            if (_handle == IntPtr.Zero)
+            {
+                throw new ObjectDisposedException(nameof(SafeLineHandle));
+            }
+
+            return _handle;
+        }
+        set
+        {
+            _handle = value;
+        }
     }
 
     /// <summary>
-    /// Release the lock on the line handle. <see cref="LibgpiodV1.gpiod_line_release"/>
+    /// Release the lock on the line handle. <see cref="Interop.LibgpiodV1.gpiod_line_release"/>
     /// </summary>
     public void ReleaseLock()
     {
-        ReleaseHandle();
+        // Contrary to intuition, this does not invalidate the handle (see comment on declaration)
+        Interop.LibgpiodV1.gpiod_line_release(_handle);
     }
 
-    public override bool IsInvalid => handle == IntPtr.Zero || handle == LibgpiodV1.InvalidHandleValue;
+    public bool IsInvalid => _handle == IntPtr.Zero || _handle == Interop.LibgpiodV1.InvalidHandleValue;
+
+    public void Dispose()
+    {
+        if (_handle != IntPtr.Zero)
+        {
+            Interop.LibgpiodV1.gpiod_line_release(_handle);
```

Comment:
Whould this call ReleaseLock instead? And should ReleaseLock also set `_handle` to IntPtr.Zero?

Reply:
They're separate because they serve different purposes. Here, we really want to close the line (from an user point of view), the other method is a special case because gpiod_line_release needs to be called before a change of the pin mode is possible. Technically, the handle is never invalidated until the owning chip is closed.